### ISSUE TITLE
Nuclei scan getting hanged

### ISSUE THREAD

**Issue by lijoabraham** [projectdiscovery/nuclei#4649]
### Nuclei version:
3.1.5

### Current Behavior:
Scans are getting hanged and getting the error.

`[ERR] Detected hanging goroutine (count=82/82) = nuclei-stacktrace-cmj9g40dlas5t54f9bj0.dump`

### Expected Behavior:
Scans should complete without any error

### Steps To Reproduce:

1. Run ' nuclei -duc -headless -or  -debug  -list "urls.txt" -w workflow_ESO.yaml -ud /app/nuclei_scan/nuclei-templates/ -rl 10 -timeout 20 -c 1 -hm -tlog -elog  -je output.json.'
2. Attached the stacktrace dump file 
[nuclei-stacktrace-cmj9g40dlas5t54f9bj0.zip](https://github.com/projectdiscovery/nuclei/files/13952182/nuclei-stacktrace-cmj9g40dlas5t54f9bj0.zip)

**Comment by tarunKoyalwar** [comment]
@lijoabraham , have you observed any other behaviour apart from `Detected hanging goroutine` . like 

1. How long was nuclei stuck ?
2. Did it exit or did you have to manually do `Ctrl + C` or kill it some other way ?


from given command you are using a custom workflow and it runs a headless template and also using `-hm`

`-hm` hang monitor is utility to detect if there was no activity in a certain goroutine for 60 sec (based on current config) and this does not always mean there is a problem ex: if network is slow or website you are using / loading takes too long to load . nuclei will wait for that much amount of time which is expected 

so unless nuclei doesn't exit or is stuck for like ~15-20 min . we can say that this is expected behaviour .

**Comment by lijoabraham** [comment]
@tarunKoyalwar Yes the command ran for ~15 min and it exited by itself. However after changing the -c to 10,  it is completing. 

I had to make the concurrency to 1 as I was getting inconsistent output while running the same command with same urls multiple times. Does -c changes make sense in this scenario?

**Comment by tarunKoyalwar** [comment]
@lijoabraham, if workflow executes more than 1 template they are executed with same concurrency as `-c <val>` flag . and inconsistent output depends on what you are trying to achieve the workflow template you are using. you have to adjust logic keeping in mind if you want to use `-c` or not . 

you can refer workflow docs for workflow templates and building conditional logic https://docs.projectdiscovery.io/templates/workflows/examples#multiple-matcher-workflow

also if command exits by itself it was most likely caused due to delayed response received from headless (since running headless templates is resource intensive and takes time to render websites)
